docs/best-practices/contributing/readme.md

```diff
@@ -0,0 +1,47 @@
+# Contributing to repositories
+
+This section summarizes best practices for contributing to code repositories.
+
+*TL;DR: Reach out to the repository owners before making a pull-request, keep contributions focussed.*
+
+### Discuss the scope of your contribution
+Whether contributing to a repository owned by your team, or to another team's product or library, the first step should
+almost always (see below for exceptions) be to reach out to the team and discuss your contribution:
+
+- Check the relevant GitHub or JIRA issues for related work
+- Send an email, comment on a relevant ticket, or discuss your contribution over a beer
+
+Small contributions like fixes for typos or handful-of-lines cosmetic changes likely don't require prior discussion.
+Just open a pull-request and comment with a friendly message of the form *"Hey @person, I am trying to fix bug #123 here
+and have also found a minor typo in the README. Let me know what you think or whether you want me to change something."*
+
+Any substantial change (including additional features to a library or service, API modifications or additions, changes
+with performance impact, additional dependencies, etc.) mandate prior discussion. This conversation is a great
+opportunity to discuss how the proposed change aligns with the team's timeline and goals, whether its architecture and
+implementation is sound, whether someone else is already working on the same issue, etc.
+
+
+### Separate refactoring steps from actual contributions
+Though this is general recommendation for all code changes, contributions to other teams' repositories in particular
+should not conflate feature/fix work with unrelated refactoring work. If your contribution requires a refactoring step,
+it is often a good idea to first submit a refactoring-only contribution, and then a follow-up contribution with the
+feature or fix.
```

Comment:
separating feature work from wiring the feature work in is also a useful way to break a change apart